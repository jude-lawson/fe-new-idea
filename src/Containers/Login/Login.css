.login-container {
  height: 100%;
  display: flex;
  align-items: center;
  justify-content: center;
  flex-direction: column;
  padding: 0 2rem;
  margin-top: -5rem;
}

.login-container--title {
  font-size: 3rem;
  color: var(--primary-white);
  font-weight: 100;
  text-align: center;
  margin-bottom: 3rem;
}

.btn-grad {
  background-image: linear-gradient(to right, #fff 51%, #fff 100%);
  padding: 1.5rem 4rem;
  border-radius: 2px;
  box-shadow: 8px 10px 20px 0 rgba(46, 61, 73, 0.15);
  transition: all 0.3s ease;
  color: var(--primary-grey);
  text-decoration: none;
}

.btn-grad:hover {
  background-position: right center;
  box-shadow: 2px 4px 8px 0 rgba(46, 61, 73, 0.2);
}
