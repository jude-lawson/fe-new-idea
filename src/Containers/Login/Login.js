import React from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import Header from '../../Containers/Header/Header';
import { firebaseLogin } from '../../thunks/auth';

import './Login.css';

export const Login = ({ firebaseLogin }) => {
  return (
    <main className="app-container">
      <Header />
      <section className="login-container">
        <h2 className="login-container--title">Oops Looks Like you need to Login</h2>
        <button className="btn-grad" onClick={() => firebaseLogin()}>Login</button>
      </section>
    </main>
  );
};

Login.propTypes = {
  firebaseLogin: PropTypes.func
};

export const mapDispatchToProps = dispatch => ({
  firebaseLogin: () => dispatch(firebaseLogin())
});

export default connect(null, mapDispatchToProps)(Login);
