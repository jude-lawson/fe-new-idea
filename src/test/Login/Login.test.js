import React from 'react';
import { shallow } from 'enzyme';
import { Login } from '../../Containers/Login/Login';

describe('<Login />', () => {
  let wrapper;

  beforeEach(() => wrapper = shallow(<Login />))

  test('should render without crashing', () => {
    expect(wrapper).toMatchSnapshot();
  })
});
