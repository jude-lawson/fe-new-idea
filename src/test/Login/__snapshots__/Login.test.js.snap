// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Login /> should render without crashing 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Unknown />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <Connect(Header) />,
        <section
          className="login-container"
>
          <h2
                    className="login-container--title"
          >
                    Oops Looks Like you need to Login
          </h2>
          <button
                    className="btn-grad"
          >
                    Login
          </button>
</section>,
      ],
      "className": "app-container",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <h2
              className="login-container--title"
>
              Oops Looks Like you need to Login
</h2>,
            <button
              className="btn-grad"
>
              Login
</button>,
          ],
          "className": "login-container",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Oops Looks Like you need to Login",
              "className": "login-container--title",
            },
            "ref": null,
            "rendered": "Oops Looks Like you need to Login",
            "type": "h2",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Login",
              "className": "btn-grad",
            },
            "ref": null,
            "rendered": "Login",
            "type": "button",
          },
        ],
        "type": "section",
      },
    ],
    "type": "main",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <Connect(Header) />,
          <section
            className="login-container"
>
            <h2
                        className="login-container--title"
            >
                        Oops Looks Like you need to Login
            </h2>
            <button
                        className="btn-grad"
            >
                        Login
            </button>
</section>,
        ],
        "className": "app-container",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <h2
                className="login-container--title"
>
                Oops Looks Like you need to Login
</h2>,
              <button
                className="btn-grad"
>
                Login
</button>,
            ],
            "className": "login-container",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Oops Looks Like you need to Login",
                "className": "login-container--title",
              },
              "ref": null,
              "rendered": "Oops Looks Like you need to Login",
              "type": "h2",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Login",
                "className": "btn-grad",
              },
              "ref": null,
              "rendered": "Login",
              "type": "button",
            },
          ],
          "type": "section",
        },
      ],
      "type": "main",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "attachTo": undefined,
    "disableLifecycleMethods": true,
    "hydrateIn": undefined,
  },
}
`;
